### PR TITLE
Shreya leaderboard unit test

### PR DESCRIPTION
**Pull request by Shreyav2000**
# Description
This pull request addresses the duplication issue within the leaderboard testing component. It includes a fix for the test organization and introduces new test cases to ensure comprehensive coverage of the LeaderBoard component's functionality.

## Related PRS (if any):
This frontend PR is  not related to the any backend PR.

…

## Main changes explained:
-Deleted the duplicate test file for the LeaderBoard component to streamline the test structure.
-Added new test cases to enhance coverage and ensure that the component's functionality is thoroughly validated.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. To perform the testing, cd into HighestGoodNetworkApp
5. Run the command npm test src/components/LeaderBoard/__tests__/Leaderboard.test.jsx
6. verify that all the test cases pass. (11/11)

## Screenshots or videos of changes:
Screenshot 2024-10-17 at 7 52 33 PM

### CODE REVIEW

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and all the test cases have passed.
Screenshot (227)

**Comment by SatyashanthiT** [comment]
I have tested this PR for the leaderboard unit test, verified the new test cases, and confirmed that all tests pass successfully. Everything is working as expected.
image

**Review by SharadhaS24 — APPROVED** [review]
Validated this PR by confirming that all 11 test cases were successfully passed.
image

File: src/components/LeaderBoard/Leaderboard.jsx
Comment (lucy-xfl) [inline src/components/LeaderBoard/Leaderboard.jsx:1]: I thought you can delete the old code directly to keep the file clean.

**Review by lucy-xfl — APPROVED** [review]
image
There is a typo in the comment that we need to run to test:  npm test src/components/leaderboard/__tests__/Leaderboard.test.jsx
11 tests passed

**Review by NK3101 — APPROVED** [review]
Tested the PR and all the test cases have passed.

**Comment by Anoushka21** [comment]
All test case passed! Its good to merge.
image

**Review by Anoushka21 — APPROVED** [review]

**Review by AshishNags — APPROVED** [review]
Approving this PR, all test cases have passed successfully.
Screenshot 2024-10-18 131317

**Comment by pallavithorat** [comment]
I have verified that all 11 test cases for Leaderboard.test.jsx passed successfully, and everything is functioning as expected.
PR#2799

**Review by CodewithKoushica — APPROVED** [review]
The removal of the duplicate test file is a good cleanup step, and the additional test cases are a solid improvement for ensuring better coverage of the LeaderBoard component. I ran the tests locally, and everything passed successfully (11/11). The organization of the tests is much cleaner now, and the added test cases effectively cover key functionality.

Great job on streamlining the test structure!

Screenshot 2024-10-18 at 10 46 12 PM

**Review by linkhanthtel — APPROVED** [review]
All 11 test cases passed as expected! Nice work!

PR #2799

**Comment by geetamatkar** [comment]
Hi @Shreyav2000 ,
Tested this PR and all 11 test cases have passed.

PR2799

**Review by ptrpengdev — APPROVED** [review]
LGTM.
Capture

**Review by nikhilgiri0226 — APPROVED** [review]
Reviewed the PR and all the 11 tests passed.

image

**Comment by samarth9201** [comment]
Verified the PR. All the Unit Test Passed.
Screenshot (87)

**Review by vishavk1992 — APPROVED** [review]
I have tested this PR. All test cases passed successfully !!

Screenshot 2024-10-21 201053

**Review by slidracoon72 — APPROVED** [review]
#2799 
@Shreyav2000 
I have reviewed and tested the PR, and all test cases have passed successfully.

Screenshot 2024-10-23 at 12 34 39 PM

**Comment by beblicarl** [comment]
All tests passed successfully

image

**Review by beblicarl — APPROVED** [review]

**Review by anirudhsk2107 — COMMENTED** [review]
I was facing this error when i ran it 
Screen Shot 2024-10-25 at 8 03 01 PM

**Review by KurtisIvey — APPROVED** [review]
Screenshot 2024-10-26 at 3 18 35 PM


Test coverage is good and seems to handle all the cases we need. Fantastic job! Well done

**Review by RachanaZha — APPROVED** [review]
I tested the PR and all the test cases have passed successfully.
Just one small suggestion: in the 'how to test' column it is written that to test the cases we have to use the command : npm test src/components/LeaderBoard/tests/Leaderboard.test.jsx..however this command doesnt work due to pattern matching issues.. i changed the path and replaced it with just the file  name - npm test Leaderboard.test.jsx ....and that worked for me.
zha_rachana_PR#2799

**Comment by mianmiantea2019** [comment]
all the test cases have passed successfully.

image

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**